### PR TITLE
[KED-2393] Update templates for lazy pipelines feature

### PR DESCRIPTION
**Pull request by lorenabalan**
~**DON'T MERGE YET.**~

Add `pipeline_registry.py` and remove `register_pipelines` hook.

Ignore CI that'll be fixed in separate PR.

### CODE REVIEW

File: pyspark-iris/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/pipeline_registry.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    """
+    data_engineering_pipeline = de.create_pipeline()
+    data_science_pipeline = ds.create_pipeline()
+
+    return {
+        "de": data_engineering_pipeline,
+        "ds": data_science_pipeline,
+        "__default__": data_engineering_pipeline + data_science_pipeline,
+    }
```
Comment (AntonyMilneQB) [inline pyspark-iris/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/pipeline_registry.py:58]:
```suggestion
    }
    
```

File: spaceflights/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/pipeline_registry.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Project pipelines."""
+from typing import Dict
+
+from kedro.pipeline import Pipeline
+
+from {{cookiecutter.python_package}}.pipelines import data_engineering as de
```
Comment (AntonyMilneQB) [inline spaceflights/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/pipeline_registry.py:34]:
```suggestion
from {{cookiecutter.python_package}}.pipelines import data_processing as dp
```

File: spaceflights/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/pipeline_registry.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    """
+    data_engineering_pipeline = de.create_pipeline()
+    data_science_pipeline = ds.create_pipeline()
+
+    return {
+        "__default__": data_engineering_pipeline + data_science_pipeline,
+        "de": data_engineering_pipeline,
+        "ds": data_science_pipeline,
+    }
```
Comment (AntonyMilneQB) [inline spaceflights/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/pipeline_registry.py:52]:
```suggestion
    data_processing_pipeline = dp.create_pipeline()
    data_science_pipeline = ds.create_pipeline()

    return {
        "__default__": data_processing_pipeline + data_science_pipeline,
        "dp": data_processing_pipeline,
        "ds": data_science_pipeline,
    }
```

**Review by AntonyMilneQB — APPROVED** [review]
Looks good, just note the changes of #28.

**Review by limdauto — APPROVED** [review]

**Review by MerelTheisenQB — APPROVED** [review]
LGTM! 👍  I am just wondering if we're changing `data engineering` to `data processing` (see Antony's comment) in the spaceflights starter, whether we should do that for all of them?

**Comment by AntonyMilneQB** [comment]
> LGTM! 👍 I am just wondering if we're changing `data engineering` to `data processing` (see Antony's comment) in the spaceflights starter, whether we should do that for all of them?

My thought on this is maybe in the future, but not now 🙂

**Pull request merged**